src/Exception/InvalidArgumentException.php

```diff
@@ -0,0 +1,14 @@
+<?php declare(strict_types=1);
+
+namespace Lmc\Matej\Exception;
+
+/**
+ * Exception represents error in the program logic.
+ */
+class InvalidArgumentException extends \InvalidArgumentException implements MatejExceptionInterface
```

Comment:
Do we need to introduce another exception type? This is IMO edge case, so it may be unnecessary to have a specific exception for this - just extending LogicException would be OK for me, and its more general, but semantically still fine (InvalidArgumentException is in SPL subtype of LogicException).